src/lib/hooks/useBlockNumber.tsx

```diff
@@ -63,6 +71,12 @@ export function BlockNumberProvider({ children }: { children: ReactNode }) {
           console.error(`Failed to get block number for chainId ${activeChainId}`, error)
         })
 
+      if (activeChainId === ChainId.MAINNET) {
+        setEthBlockNumber(block)
+      } else {
+        RPC_PROVIDERS[ChainId.MAINNET].getBlockNumber().then(setEthBlockNumber)
```

Comment:
any reason not to use async await here?

Reply:
We could but then we need to make the entire useEffect function async. Also I was just following the structure used above on line 65